projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py

```diff
@@ -54,6 +54,32 @@ def new_datajob(
         return DataJob(data_job_directory, core_context, name)
 
 
+def save_step_result(context: JobContext, result: StepResult):
+    step_results = context.core_context.state.get(CommonStoreKeys.STEP_RESULTS)
+    step_results.append(result)
+    context.core_context.state.set(CommonStoreKeys.STEP_RESULTS, step_results)
+    return result
+
+
+def save_execution_result(
+    context: JobContext, start_time=None, exception=None, blamee=None
+):
+    execution_status = context.core_context.state.get(CommonStoreKeys.EXECUTION_STATUS)
+    step_results = context.core_context.state.get(CommonStoreKeys.STEP_RESULTS)
+    execution_result = ExecutionResult(
+        context.name,
+        context.core_context.state.get(CommonStoreKeys.EXECUTION_ID),
+        start_time,
+        datetime.utcnow(),
+        execution_status,
+        step_results,
```

Comment:
We keep step_results in 2 places. 


Why not build ExecutionResult over time. 

when job starts we add it almost empty -just name, id, start time ,  
you can add the steps as they happen ?